typing_extensions/src/test_typing_extensions.py

```diff
@@ -1681,6 +1682,17 @@ def test_keys_inheritance(self):
             'voice': str,
         }
 
+    def test_is_typeddict(self):
+        assert is_typeddict(Point2D) is True
+        assert is_typeddict(Union[str, int]) is False
+        # classes, not instances
+        assert is_typeddict(Point2D()) is False
+
+    @skipUnless(TYPING_3_8_0, "Python 3.8+ required")
+    def test_is_typeddict_against_typeddict_from_typing(self):
+        Point = typing.TypedDict('Point', {'x': int, 'y': int})
```

Comment:
Please also test (for both the typing and typing_extensions versions):
- a class-based TypedDict
- a class inheriting from another TypedDict